### PR TITLE
Markmenship Hunter Initial Setup

### PR DESCRIPTION
**Pull request by Jlassie82**
Added spec basics for Marksmenship Hunter: Alwaysbecasting, CastEfficiency, CooldownTracker & DamageDone

### CODE REVIEW

File: src/Parser/MarksmanshipHunter/CombatLogParser.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const results = super.generateResults();
+
+    this.selectedCombatant._combatantInfo.gear.forEach(function (value) {
+      const equippedItem = ITEMS[value.id];
+
+      if (equippedItem !== undefined && equippedItem.quality === 5) {
+        results.items.push({
+          item: equippedItem,
+          result: (
+            <dfn data-tip="">
```
Comment (MartijnHols) [inline src/Parser/MarksmanshipHunter/CombatLogParser.js]: This does nothing and can be omitted.

File: src/Parser/MarksmanshipHunter/CombatLogParser.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    cooldownTracker: CooldownTracker,
+  };
+
+  generateResults() {
+    const results = super.generateResults();
+
+    this.selectedCombatant._combatantInfo.gear.forEach(function (value) {
+      const equippedItem = ITEMS[value.id];
+
+      if (equippedItem !== undefined && equippedItem.quality === 5) {
```
Comment (MartijnHols) [inline src/Parser/MarksmanshipHunter/CombatLogParser.js]: The magic number `5` should be replaced with a constant, you can find item quality constants in the `common/ITEM_QUALITIES` file.

File: src/Parser/MarksmanshipHunter/Modules/Features/AlwaysBeCasting.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+  ];
+
+  suggestions(when) {
+    const deadTimePercentage = this.totalTimeWasted / this.owner.fightDuration;
+
+    when(deadTimePercentage).isGreaterThan(0.2)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your dead GCD time can be improved. Try to Always Be Casting (ABC), try to reduce the delay between casting spells. Even if you have to move, try casting something instant with range like <SpellLink id={SPELLS.FLAMETONGUE.id}/> or <SpellLink id={SPELLS.ROCKBITER.id}/></span>)
```
Comment (MartijnHols) [inline src/Parser/MarksmanshipHunter/Modules/Features/AlwaysBeCasting.js]: These don't seem like things a Hunter can do. ;) What *Hunter* spells would you recommend casting while moving?

**Comment by MartijnHols** [comment]
Thanks! See the comments for some feedback.

File: src/Parser/MarksmanshipHunter/Modules/Features/AlwaysBeCasting.js (hunk 3 above)
Comment (Jlassie82) [reply src/Parser/MarksmanshipHunter/Modules/Features/AlwaysBeCasting.js]: yeah, used some of nighteyz code as a template and missed this. Will correct in next pull request

**Pull request merged**